nexus/reconfigurator/blippy/src/checks.rs

```diff
@@ -314,6 +314,10 @@ impl<'a> DatasetsBySledCache<'a> {
 
         let mut by_zpool = BTreeMap::new();
         for dataset in sled_config.datasets.iter() {
+            if dataset.disposition.is_expunged() {
```

Comment:
Without this change, `test_update_internal_dns` fails.

I believe that blippy observes two datasets in the blueprint - one expunged, one not expunged - and considered them duplicates.

With this change, the expunged one is ignored.